learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -136,9 +218,23 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
     if (content == null) {
       return;
     }
-    final taskSnippetId = content.taskSnippetId;
-    await _setPlaygroundSnippet(taskSnippetId);
-    _isShowingSolution = false;
+    await _setCurrentSnippet();
+  }
+
+  Future<void> _setCurrentSnippet() async {
+    final unit = _currentUnitContent;
+    if (unit != null) {
+      final String? snippetId;
+      if (_resetSnippet) {
```

Comment:
The ability to reset is secondary to other state. It should not be checked when deciding what to load.

Reply:
Should `isSnippetReset` be passed as an argument into `_setCurrentSnippet`?

Reply:
Deleted.